src/overlays/actors/ovl_Arms_Hook/z_arms_hook.c

```diff
@@ -23,27 +29,329 @@ const ActorInit Arms_Hook_InitVars = {
+        sp60.x = this->unk_1F4.x - (this->unk_1E8.x - this->unk_1F4.x);
+        sp60.y = this->unk_1F4.y - (this->unk_1E8.y - this->unk_1F4.y);
+        sp60.z = this->unk_1F4.z - (this->unk_1E8.z - this->unk_1F4.z);
+        if (func_8003DE84(&globalCtx->colCtx, &sp60, &this->unk_1E8, &sp78, &poly, 1, 1, 1, 1, &dynaPolyID) != 0) {
+            if (func_8002F9EC(globalCtx, &this->actor, (UNK_TYPE)poly, dynaPolyID, (UNK_TYPE)&sp78) == 0) {
+                sp5C = poly->norm.x * S16_MAX_RECIPROCAL;
+                sp58 = poly->norm.z * S16_MAX_RECIPROCAL;
+                Math_Vec3f_Copy(curPos, &sp78);
+                this->actor.posRot.pos.x += 10.0f * sp5C;
+                this->actor.posRot.pos.z += 10.0f * sp58;
+                this->unk_210 = 0;
+                if (func_80041FE8(&globalCtx->colCtx, poly, dynaPolyID) != 0) {
+                    if (dynaPolyID != 0x32) {
+                        dynaPolyActor = DynaPolyInfo_GetActor(&globalCtx->colCtx, dynaPolyID);
+                        if (dynaPolyActor != NULL) {
+                            ArmsHook_AttachHookToActor(this, &dynaPolyActor->actor);
+                        }
+                    }
+                    func_80865044(this);
+                    Audio_PlaySoundGeneral(NA_SE_IT_HOOKSHOT_STICK_OBJ, &this->actor.unk_E4, 4, &D_801333E0,
+                                           &D_801333E0, &D_801333E8);
+                    return;
+                }
+                func_80062D60(globalCtx, curPos);
+                Audio_PlaySoundGeneral(NA_SE_IT_HOOKSHOT_REFLECT, &this->actor.unk_E4, 4, &D_801333E0, &D_801333E0,
+                                       &D_801333E8);
+                return;
+            }
+        }
+        if ((globalCtx->state.input[0].press.in.button &
+             (A_BUTTON | B_BUTTON | R_TRIG | U_CBUTTONS | L_CBUTTONS | R_CBUTTONS | D_CBUTTONS)) != 0) {
+            this->unk_210 = 0;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Arms_Hook/func_8086518C.s")
+void ArmsHook_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ArmsHook* this = THIS;
+    this->actionFunc(this, globalCtx);
+    this->unk_1F4 = this->unk_1E8;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Arms_Hook/ArmsHook_Update.s")
+void ArmsHook_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    ArmsHook* this = THIS;
+    u32 pad3;
+    Player* player = PLAYER;
+    Vec3f sp78;
+    Vec3f sp6C;
+    Vec3f sp60;
+    f32 sp5C;
+    f32 sp58;
+    GraphicsContext* gfxCtx;
+    Gfx* dispRefs[4];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Arms_Hook/ArmsHook_Draw.s")
+    if (player->actor.draw != NULL) {
+        if (player->unk_15D == 0xF) {
+            gfxCtx = globalCtx->state.gfxCtx;
+            Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_arms_hook.c", 850);
+            if (((ArmsHookActionFunc)&func_8086518C != this->actionFunc) || (this->unk_210 <= 0)) {
```

Comment:
cast shouldnt be needed I would think. they are both ArmsHookActionFuncs. and I would put this->actionFunc on the left side of the comparison if it works but I know it affects codegen sometimes